constraint-solver/src/constraint_system.rs

```diff
@@ -84,6 +84,10 @@ impl<T: RuntimeConstant, V> ConstraintSystem<T, V> {
             .extend(system.algebraic_constraints);
         self.bus_interactions.extend(system.bus_interactions);
     }
+
+    pub fn algebraic_constraints(&self) -> &[GroupedExpression<T, V>] {
```

Comment:
The field `algebraic_constraints` is pub so I don't think we need a function here

Reply:
I did this to avoid changing all the call sites. But it's also safer, since write access to the constraints could invalidate the occurence map. Ideally I think the constraints should not be pub.

Reply:
There is not occurrence map in the basic constraint system, that's why it's safe to make the constraints `pub`

Reply:
oh good point, wrong file! thanks